localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/http.py

```diff
@@ -1,20 +1,143 @@
+from http import HTTPMethod
+from typing import Optional, TypedDict
+
+import requests
+from werkzeug.datastructures import Headers
+
+from localstack.aws.api.apigateway import Integration
+from localstack.http import Response
+
+from ..context import IntegrationRequest, InvocationRequest, RestApiInvocationContext
+from ..helpers import render_uri_with_path_parameters
 from .core import RestApiIntegration
 
+NO_BODY_METHODS = {
+    HTTPMethod.OPTIONS,
+    HTTPMethod.GET,
+    HTTPMethod.HEAD,
+}
+
+
+class SimpleHttpRequest(TypedDict, total=False):
+    method: HTTPMethod | str
+    url: str
+    params: Optional[dict[str, str | list[str]]]
+    data: bytes
+    headers: Optional[dict[str, str]]
+    cookies: Optional[dict[str, str]]
+    timeout: Optional[int]
+    allow_redirects: Optional[bool]
+    stream: Optional[bool]
+    verify: Optional[bool]
+    # TODO: check if there was a situation where we'd pass certs?
+    cert: Optional[str | tuple[str, str]]
+
+
+class BaseRestApiHttpIntegration(RestApiIntegration):
+    @staticmethod
+    def _get_default_headers(context: RestApiInvocationContext) -> dict[str, str]:
+        # passing full context as the trace-id will probably also come from it
+        # TODO: get right casing in case we can override them with mappings/override?
+        return {
+            "x-amzn-apigateway-api-id": context.api_id,
+            "x-amzn-trace-id": "",  # TODO
+            "user-agent": f"AmazonAPIGateway_{context.api_id}",
+        }
 
-class RestApiHttpIntegration(RestApiIntegration):
+
+class RestApiHttpIntegration(BaseRestApiHttpIntegration):
     """
     This is a REST API integration responsible to send a request to another HTTP API.
     https://docs.aws.amazon.com/apigateway/latest/developerguide/setup-http-integrations.html#api-gateway-set-up-http-proxy-integration-on-proxy-resource
     """
 
     name = "HTTP"
 
+    def invoke(self, context: RestApiInvocationContext) -> Response:
+        integration_req: IntegrationRequest = context.integration_request
+        method = integration_req["http_method"]
+
+        default_apigw_headers = self._get_default_headers(context)
+        default_apigw_headers.update(integration_req["headers"])
+
+        request_parameters: SimpleHttpRequest = {
+            "method": method,
+            "url": integration_req["uri"],
+            "params": integration_req["query_string_parameters"],
+            "headers": default_apigw_headers,
+        }
+
+        if method not in NO_BODY_METHODS:
+            request_parameters["data"] = integration_req["body"]
+
+        # TODO: configurable timeout (29 by default)
+        # request_parameters["timeout"] = 29
```

Comment:
Question: Do you know if the timeout also affects proxy integration? Is it something we should manage at the Integration handler level?

Reply:
I believe this should be the value `timeoutInMillis` from the `Integration`: https://docs.aws.amazon.com/apigateway/latest/api/API_Integration.html

So it should be managed at the integration level, I also had it for `PROXY` I think but removed it when refactoring. I'll update 👍 

Thinking of which, there's also caching which is currently not supported at all, we might want to think of that later for the features not currently implemented!